slatedb/src/db.rs

```diff
@@ -77,30 +92,53 @@ impl DbInner {
         clock: Arc<dyn Clock + Send + Sync>,
         table_store: Arc<TableStore>,
         manifest: DirtyManifest,
-        wal_flush_notifier: UnboundedSender<WalFlushMsg>,
         memtable_flush_notifier: UnboundedSender<MemtableFlushMsg>,
         write_notifier: UnboundedSender<WriteBatchMsg>,
         stat_registry: Arc<StatRegistry>,
     ) -> Result<Self, SlateDBError> {
+        // both last_seq and last_committed_seq will be updated after WAL replay.
+        let last_l0_seq = manifest.core.last_l0_seq;
+        let last_seq = Arc::new(MonotonicSeq::new(last_l0_seq));
+        let last_committed_seq = Arc::new(MonotonicSeq::new(last_l0_seq));
+        let last_remote_persisted_seq = Arc::new(MonotonicSeq::new(last_l0_seq));
+
         let mono_clock = Arc::new(MonotonicClock::new(clock, manifest.core.last_l0_clock_tick));
+
+        // state are mostly manifest, including IMM, L0, etc.
         let state = Arc::new(RwLock::new(DbState::new(manifest)));
+
         let db_stats = DbStats::new(stat_registry.as_ref());
         let wal_enabled = DbInner::wal_enabled_in_options(&settings);
 
         let reader = Reader {
             table_store: Arc::clone(&table_store),
             db_stats: db_stats.clone(),
             mono_clock: Arc::clone(&mono_clock),
-            wal_enabled,
+            last_committed_seq: Arc::clone(&last_committed_seq),
```

Comment:
Curious why the `&` ref here.

Reply:
it followed the previous style in `table_store`, `mono_clock` etc. i feel using `.clone()` is better, let me change them together